hpx/runtime/resource/detail/partitioner.hpp

```diff
@@ -53,7 +54,8 @@ namespace hpx { namespace resource { namespace detail
 
     private:
         init_pool_data(const std::string &name,
-            scheduling_policy = scheduling_policy::unspecified);
+            scheduling_policy = scheduling_policy::unspecified,
+            hpx::util::optional<hpx::threads::policies::scheduler_mode> = hpx::util::nullopt);
```

Comment:
Wouldn't it be simpler to define an enumerator value representing a 'default_mode'?